tests/test-dladdr.c

```diff
@@ -0,0 +1,163 @@
+/* required for non Windows builds, must be set in front of the first system include */
```

Comment:
I would suggest to put at the beggining of this file that on Linux this test should be compiled with parameters `-Wl,--export-dynamic -pie`. Otherwise symbol for `main` would not resolved by glibc's `dladdr` implementation and test would fail.